Dockerfile

```diff
@@ -12,24 +12,24 @@
 # the License.
 #
 
-FROM adoptopenjdk/openjdk11:alpine
+FROM eclipse-temurin:11-jdk-alpine as builder
+WORKDIR /build
+RUN apk add --no-cache git
+COPY ./ ./
+# if the disitrbution is provided, do nothing - else build it
+RUN if [[ ! -d thirdeye-distribution/target/thirdeye-distribution-*-dist/thirdeye-distribution-* ]]; then ./mvnw package -U -DskipTests; fi
 
-EXPOSE 8080
-EXPOSE 8081
-EXPOSE 8443
-
-RUN addgroup -S thirdeye && \
-  adduser -S thirdeye -G thirdeye && \
-  chown thirdeye:thirdeye /home/thirdeye
+FROM eclipse-temurin:11-jre-alpine
+RUN addgroup -g 1000 thirdeye && \
+  adduser -u 1000 thirdeye -G thirdeye -D
 
 USER thirdeye
 WORKDIR /home/thirdeye
 
-COPY thirdeye-distribution/target/thirdeye-distribution-*-dist.tar.gz thirdeye-distribution-dist.tar.gz
-
-RUN mkdir thirdeye && tar -xvf thirdeye-distribution-dist.tar.gz -C thirdeye --strip-components=1 && \
-  rm thirdeye-distribution-dist.tar.gz
+EXPOSE 8080
+EXPOSE 8081
+EXPOSE 8443
 
-WORKDIR /home/thirdeye/thirdeye
```

Comment:
sorry I missed this previously, 
can we keep this workdir? 

agree a workdir `app` or `home/thirdeye` is better than `home/thirdeye/thirdeye` but changing the workdir would break the kubernetes setup.
Eg it would break here: https://github.com/startreedata/thirdeye/blob/46bf2eaec13bb1399dd7cdd9edd66c0d82930460/kubernetes/helm/startree-thirdeye/templates/coordinator/deployment.yaml#L122 (and in many other places)